cns/ipampoolmonitor/ipampoolmonitor.go

```diff
@@ -11,121 +13,193 @@ import (
 )
 
 var (
-	increasePoolSize = 1
-	decreasePoolSize = -1
-	doNothing        = 0
+	increasePoolSize = int64(1)
+	decreasePoolSize = int64(-1)
+	doNothing        = int64(0)
 )
 
 type CNSIPAMPoolMonitor struct {
-	initialized bool
+	initialized    bool
+	pendingRelease bool
+
+	cachedSpec               nnc.NodeNetworkConfigSpec
+	cachedSecondaryIPConfigs map[string]cns.SecondaryIPConfig
 
 	cns            cns.HTTPService
 	rc             requestcontroller.RequestController
 	scalarUnits    cns.ScalarUnits
-	MinimumFreeIps int
-	MaximumFreeIps int
+	MinimumFreeIps int64
+	MaximumFreeIps int64
 
 	sync.RWMutex
```

Comment:
Consider naming this mutex instead of embedding it. Embedding it makes the lock/unlock methods exported as well, which is something calling code should not be aware of.